src/main/java/io/pakland/mdas/githubstats/domain/entity/UserReviewAggregation.java

```diff
@@ -1,30 +1,77 @@
 package io.pakland.mdas.githubstats.domain.entity;
 
+import java.util.ArrayList;
 import java.util.List;
 
 public class UserReviewAggregation implements CSVExportable {
 
-    private int commentLengthSum = 0;
-    private int commentCount = 0;
+    private int internalCommentLengthSum = 0;
+    private int externalCommentLengthSum = 0;
+    private int internalCommentCount = 0;
+    private int externalCommentCount = 0;
 
     public UserReviewAggregation aggregate(List<UserReview> userReviews) {
-        commentCount = userReviews.size();
-        commentLengthSum = userReviews.stream().mapToInt(UserReview::sumCommentLength).sum();
+        List<UserReview> internalReviews = new ArrayList<>();
+        List<UserReview> externalReviews = new ArrayList<>();
+
+        userReviews.forEach(userReview -> {
```

Comment:
I don't think is needed to store filtered by isInternal lists, just its comment length sum. This way you avoid going over the list two times ( you avoid to userReviews + internalReviews + externalReviews )

```java
userReviews.forEach(userReview -> {
    if (userReview.isInternal()) {
        internalCommentLengthSum += userReview.sumCommentLength();
        internalCommentCount++;
    } else {
        externalCommentLengthSum += userReview.sumCommentLength();
        externalCommentCount++;
    }
});
```

Reply:
Thanks for the feedback. The latest commit includes your suggested changes now.